src/rsz/src/Rebuffer.cc

```diff
@@ -109,51 +109,78 @@ void pruneCapVsSlackOptions(StaState* sta, BufferedNetSeq& options)
                             const BufferedNetPtr& choice,
                             int& buffer_count,
                             int& load_count,
                             int& wire_count,
-                            int& junction_count)
+                            int& junction_count,
+                            std::set<odb::dbModule*>& load_modules)
 {
   switch (choice->type()) {
     case BufferedNetType::buffer: {
       buffer_count++;
-      characterizeChoiceTree(level + 1,
+      characterizeChoiceTree(nwk,
+                             level + 1,
                              choice->ref(),
                              buffer_count,
                              load_count,
                              wire_count,
-                             junction_count);
+                             junction_count,
+                             load_modules);
       break;
     }
     case BufferedNetType::wire: {
       wire_count++;
-      characterizeChoiceTree(level + 1,
+      characterizeChoiceTree(nwk,
+                             level + 1,
                              choice->ref(),
                              buffer_count,
                              load_count,
                              wire_count,
-                             junction_count);
+                             junction_count,
+                             load_modules);
       break;
     }
     case BufferedNetType::junction: {
       junction_count++;
-      characterizeChoiceTree(level + 1,
+      characterizeChoiceTree(nwk,
+                             level + 1,
                              choice->ref(),
                              buffer_count,
                              load_count,
                              wire_count,
-                             junction_count);
-      characterizeChoiceTree(level + 1,
+                             junction_count,
+                             load_modules);
+      characterizeChoiceTree(nwk,
+                             level + 1,
                              choice->ref2(),
                              buffer_count,
                              load_count,
                              wire_count,
-                             junction_count);
+                             junction_count,
+                             load_modules);
       break;
     }
     case BufferedNetType::load: {
+      const Pin* load_pin = choice->loadPin();
+      odb::dbITerm* load_iterm = nullptr;
+      odb::dbBTerm* load_bterm = nullptr;
+      odb::dbModITerm* load_moditerm = nullptr;
+
+      nwk->staToDb(load_pin, load_iterm, load_bterm, load_moditerm);
+      if (load_iterm) {
+        dbInst* load_inst = load_iterm->getInst();
+        if (load_inst) {
+          auto top_module = nwk->block()->getTopModule();
+          if (load_inst->getModule() == top_module) {
+            load_modules.insert(top_module);
+          } else {
+            load_modules.insert(load_inst->getModule());
+          }
```

Comment:
I don't see the need for the if/else, it looks like both branches just do `load_modules.insert(load_inst->getModule());`